pep517/build.py

```diff
@@ -3,25 +3,50 @@
 import argparse
 import logging
 import os
-import contextlib
 import pytoml
 import shutil
-import errno
-import tempfile
 
 from .envbuild import BuildEnvironment
 from .wrappers import Pep517HookCaller
+from .dirtools import tempdir, mkdir_p
 
 log = logging.getLogger(__name__)
 
 
-@contextlib.contextmanager
-def tempdir():
-    td = tempfile.mkdtemp()
+def validate_system(system):
+    """
+    Ensure build system has the requisite fields.
+    """
+    required = {'requires', 'build-backend'}
+    if required > set(system):
+        missing = required - set(system)
+        message = "Missing required fields: {missing}".format(**locals())
```

Comment:
I'd rather lose the `locals()` introspection trick. It's a distraction if you're not used to reading it, and it doesn't save any characters.

Reply:
I prefer this technique because it most closely translates to the syntax for f-strings, which I'd use here once available. I see the `**locals()` trick as a wart that's a valuable reminder to use f-strings. But since you feel strongly enough, I can omit it.

Reply:
Thanks; I understand the point, but I think the downside of the wart outweighs the reminder to use f-strings, which aren't a big win here anyway.

Reply:
Fixed in bdad30a.